### ISSUE TITLE
Track merged patches by patch-id

### ISSUE THREAD

**Issue by bonzini** [patchew-project/patchew#52]
`git patch-id` computes a hash of the patch contents, and is able to track merged patches even if they are not applied with a Message-id trailer (patchwork does this).

**Comment by famz** [comment]
Thanks for the idea! Over other mechanisms, "git patch-id" has the advantage that it doesn't have false positives. What I've had in mind for a while as an alternative is `(subject, author)` matching since it is a common practice that maintainers can make small adjustments to the diff, or do trivial but manual rebase, resolving some conflicts. I think "git patch-id" will not be enough in these cases? Of course, on that basis, any matching algorithm will not cover 100% cases.

So, IIUC, this feature means we add a `patch_ids` parameter to the `update-project-head` API method, and a `patch_id` parameter to `Message` model for any `is_patch=true` items. A migration can initialize the old data.

**Comment by bonzini** [comment]
Yeah, perhaps (subject, author) is simpler and easier, and the data is already there without need for a migration.
